Code improvement

Martinski4GitHub opened the pull request:
1) FIXED: Modified code to get the full current F/W version installed for comparison purposes because the latest F/W builds "3006.102.x.y" were generating wrong results when compared with older "3004.386.x.y" and "3004.388.x.y" F/W versions.

2) IMPROVED: Modified code so that every time the SSH CLI menu is run, it checks if the WebGUI page has already been mounted. If not found mounted, the script will run the code to remount the WebGUI.

3) IMPROVED: Added "export PATH" statement to give the built-in binaries higher priority than the equivalent Entware binaries.

4) Removed/modified code related to "var $j = jQuery.noConflict();" which is now considered obsolete.

Martinski4GitHub committed:
Code improvement

Removed/modified code related to "var $j = jQuery.noConflict();" which is now considered obsolete.

Martinski4GitHub committed:
Merge branch 'jackyaz:develop' into develop

jackyaz commented:
this was needed on older firmwares due to old jquery. does this affect minimum version support?

Martinski4GitHub commented:
>
> this was needed on older firmwares due to old jquery. does this affect minimum version support?
>
>
Sorry, I don't really understand the question about "minimum version support" WRT jQuery.

My current understanding is as follows. According to dave14305, the changes related to the line "var $j=jQuery.noConflict();" were made to ensure compatibility & avoid conflicts with John's forked F/W builds. However, John (aka john9527) has essentially stopped the development of his fork. The last official release was issued on 2021-Dec-19, and the last Development-Beta release was issued on 2022-Aug-03. So for all intents and purposes, John's fork has been abandoned for at least 2 years now, and no one else is stepping up to take his place. So the current recommendation is to remove those "`$j`" changes since they are now considered obsolete and are starting to cause some issues with the latest RMerlin F/W builds.

I hope this helps to provide more context.

Martinski4GitHub committed:
Fixes & Improvements

1) FIXED: Modified code to get the full current F/W version installed for comparison purposes because the latest F/W builds "3006.102.x.y" were generating wrong results when compared with older "3004.386.x.y" and "3004.388.x.y" F/W versions.

2) IMPROVED: Modified code so that every time the SSH CLI menu is run, it checks if the WebGUI page has already been mounted. If not found mounted, the script will run the code to remount the WebGUI.

3) IMPROVED: Added "export PATH" statement to give the built-in binaries higher priority than the equivalent Entware binaries.

Martinski4GitHub commented:
@jackyaz,

This PR now includes a fix to properly compare F/W versions using the full version string (e.g. "**3004.386.4.2**") instead of the short version (e.g. "**386.4.2**") because with the latest **3006.102.x.y** release builds, the version comparisons are giving erroneous results (e.g. "**102.x.y**" is less than "**386.4.2**").

Whenever you have a chance, please merge this PR into your 'develop' branch.

Thank you.

Martinski4GitHub committed:
Minor Improvements.

Simple minor improvements.

helpfulBro commented:
@Martinski4GitHub - does this still work on 3004.388.x versions?

Martinski4GitHub commented:
> @Martinski4GitHub - does this still work on 3004.388.x versions?
>
If by "this" you're referring specifically to the changes included in this PR then yes, they are backward compatible with "3004.388.x.y" F/W versions.

Martinski4GitHub commented:
Jack Yaz,
Thank you for taking the time to merge the pending PRs.
Have a great weekend, mate!!
